cl/corpus_importer/management/commands/financial_disclosures.py

```diff
@@ -0,0 +1,344 @@
+        return True
+    return False
+
+
+def extract_content(pdf_bytes: bytes) -> Dict:
+    """Extract the content of the PDF.
+
+    Attempt extraction using multiple methods if necessary.
+
+    :param pdf_bytes: The byte array of the PDF
+    :return:The extracted content
+    """
+    logger.info("Beginning Extraction")
+
+    # Extraction takes between 7 seconds and 80 minutes for super
+    # long Trump extraction with ~5k investments
+    extractor_response = requests.post(
+        settings.BTE_URLS["extract-disclosure"],
+        files={"pdf_document": ("file", pdf_bytes)},
+        timeout=60 * 120,
+    )
+
+    if (
+        extractor_response.status_code != 200
+        or extractor_response.json()["success"] is False
+    ):
+        # Try second method
+        logger.info("Attempting second extraction")
+        extractor_response = requests.post(
+            settings.BTE_URLS["extract-disclosure-jw"],
+            files={"file": ("file", pdf_bytes)},
+            timeout=60 * 60,
+        )
+
+        if (
+            extractor_response.status_code != 200
+            or extractor_response.json()["success"] is False
+        ):
+            logger.info("Could not extract data from this document")
+            return {}
+
+    logger.info("Processing extracted data")
+    return extractor_response.json()
+
+
+def get_report_type(extracted_data: dict) -> int:
+    """Get report type if available
+
+    :param extracted_data: Document information
+    :return: Disclosure type
+    """
+    if extracted_data.get("initial"):
+        return FinancialDisclosure.INITIAL
+    elif extracted_data.get("nomination"):
+        return FinancialDisclosure.NOMINATION
+    elif extracted_data.get("annual"):
+        return FinancialDisclosure.ANNUAL
+    elif extracted_data.get("final"):
+        return FinancialDisclosure.FINAL
+
+
+def save_disclosure(
+    extracted_data: dict, disclosure: FinancialDisclosure
+) -> None:
+    """Save financial data to system.
+
+    Wrapped in a transaction, we fail if anything fails.
+
+    :param disclosure: Financial disclosure
+    :param extracted_data: disclosure
+    :return:None
+    """
+    addendum = "Additional Information or Explanations"
+
+    # Process and save our data into the system.
+    with transaction.atomic():
+        disclosure.has_been_extracted = True
+        disclosure.addendum_content_raw = extracted_data[addendum]["text"]
+        disclosure.addendum_redacted = extracted_data[addendum]["is_redacted"]
+        disclosure.is_amended = extracted_data.get("amended") or False
+        disclosure.report_type = get_report_type(extracted_data)
+        disclosure.save()
+
+        for investment in extracted_data["sections"]["Investments and Trusts"][
+            "rows"
+        ]:
+            Investment.objects.create(
+                financial_disclosure=disclosure,
+                redacted=True
+                in [v["is_redacted"] for _, v in investment.items()],
```

Comment:
Would this be simpler here and elsewhere:

```suggestion
                redacted=any(v["is_redacted"] for v in investment.values()),
```

Reply:
√